plugin/storage/cassandra/factory.go

```diff
@@ -128,34 +133,20 @@ func (f *Factory) InitFromViper(v *viper.Viper, _ *zap.Logger) {
 // InitFromOptions initializes factory from options.
 func (f *Factory) configureFromOptions(o *Options) {
 	f.Options = o
-	// TODO this is a hack because we do not define defaults in Options
-	if o.others == nil {
-		o.others = make(map[string]*NamespaceConfig)
-	}
-	f.primaryConfig = o.GetPrimary()
-	f.archiveConfig = f.Options.Get(archiveStorageConfig)
+	f.config = o.GetPrimary()
```

Comment:
is GetPrimary the right name? Or should it become CreateConfig?

Reply:
changed to `GetConfig`